drivers/SmartThings/zigbee-window-treatment/src/test/test_zigbee_window_aqara_curtain.lua

```diff
@@ -0,0 +1,346 @@
+    )
+  end
+)
+
+test.register_coroutine_test(
+  "Window Shade state open",
+  function()
+    test.socket.capability:__set_channel_ordering("relaxed")
+    test.socket.zigbee:__queue_receive(
+      {
+        mock_device.id,
+        WindowCovering.attributes.CurrentPositionLiftPercentage:build_test_attr_report(mock_device, 100)
+      }
+    )
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("main", capabilities.windowShade.windowShade.open())
+    )
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("main", capabilities.windowShadeLevel.shadeLevel(100))
+    )
+  end
+)
+
+test.register_coroutine_test(
+  "Window Shade state partially open",
+  function()
+    test.socket.capability:__set_channel_ordering("relaxed")
+    test.socket.zigbee:__queue_receive(
+      {
+        mock_device.id,
+        WindowCovering.attributes.CurrentPositionLiftPercentage:build_test_attr_report(mock_device, 25)
+      }
+    )
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("main", capabilities.windowShade.windowShade.partially_open())
+    )
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("main", capabilities.windowShadeLevel.shadeLevel(25))
+    )
+  end
+)
+
+test.register_coroutine_test(
+  "WindowShade open cmd handler",
+  function()
+    test.socket.capability:__queue_receive(
+        {
+          mock_device.id,
+          { capability = "windowShade", component = "main", command = "open", args = {} }
+        }
+    )
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("main", capabilities.windowShade.windowShade.closed())
+    )
+    test.socket.zigbee:__expect_send({
+      mock_device.id,
+      WindowCovering.server.commands.GoToLiftPercentage(mock_device, 100)
+    })
+    test.wait_for_events()
+  end
+)
+
+test.register_coroutine_test(
+  "WindowShade close cmd handler",
+  function()
+    test.socket.capability:__queue_receive(
+        {
+          mock_device.id,
+          { capability = "windowShade", component = "main", command = "close", args = {} }
+        }
+    )
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("main", capabilities.windowShade.windowShade.closed())
+    )
+    test.socket.zigbee:__expect_send({
+      mock_device.id,
+      WindowCovering.server.commands.GoToLiftPercentage(mock_device, 0)
+    })
+    test.wait_for_events()
+  end
+)
+
+test.register_coroutine_test(
+  "WindowShade pause cmd handler",
+  function()
+    test.socket.capability:__queue_receive(
+        {
+          mock_device.id,
+          { capability = "windowShade", component = "main", command = "pause", args = {} }
+        }
+    )
+    test.socket.capability:__expect_send(
+      mock_device:generate_test_message("main", capabilities.windowShade.windowShade.closed())
```

Comment:
Does `pause` always result in a `closed` event?  otherwise is there another case that isn't being tested where a `pause` could result in an `open` or `partially_open`?

Reply:
@sbkimaqara
You can expand the coverage by setting the 'SHADE_LEVEL' explicitly like below...
ex)
local SHADE_LEVEL = "shadeLevel"
test.register_coroutine_test(
  "WindowShade pause cmd handler(partially open)",  -- open /closed
  function()
    mock_device:set_field(SHADE_LEVEL, 50) -- 0, 100
    test.socket.capability:__queue_receive(
        {
          mock_device.id,
          { capability = "windowShade", component = "main", command = "pause", args = {} }
        }
    )
    test.socket.capability:__expect_send(
      mock_device:generate_test_message("main", capabilities.windowShade.windowShade.partially_open()) -- closed() , open()
    )
    test.socket.zigbee:__expect_send({
      mock_device.id,
      WindowCovering.server.commands.Stop(mock_device)
    })
    test.wait_for_events()
  end
)

Reply:
Thank you for your advice. I updated code.